### DISCUSSION TITLE
DataSearch::customQuery wrapped in multiple queries

### DISCUSSION THREAD

**Discussion by rs-unsilo** [appbaseio/reactivesearch#2135] [Q&A]
Hi list

We are using the customQuery function in the DataSearch component. It seems the query you supply is wrapped in a lot of layers. Is it possible to avoid this?

`    "query": {
        "bool": {
            "must": [
                {
                    "bool": {
                        "must": [
                            {
                                >>>> Custom query here <<<<<<
                            }
                        ]
                    }
                }
            ]
        }
    },
`

**Comment by siddharthlatest** [comment]
@rs-unsilo What are you intending to do with the `customQuery` - seems you're taking the existing query and then tweaking it?

**Reply by rs-unsilo** [reply]
I don't understand the question. I don't consider it tweaking I'm just using my own querystring which as I read the documentation it's supposed to work with:

"customQuery Function takes value and props as parameters and returns the data query to be applied to the component, as defined in Elasticsearch Query DSL. Note: customQuery is called on value changes in the DataSearch component as long as the component is a part of react dependency of at least one other component."

It works but the query I build is wrapped in all these extra enclosing layers which don't really make sense to me.
